index.js

```diff
@@ -648,3 +758,467 @@ function validateColors() {
   dialog_color.dialog('close')
   updateOutput()
 }
+
+/// When button import is clicked
+$('#import').click(function () {
+  // Animation prevented, set display to inline-block
+  $('#import-field').css('display', 'inline-block')
+
+  var success = true
+
+  if ($('#import-field').hasClass('visible') === true) {
+    // Try to import
+    if ($('#import-field').val().length > 0) {
+      try {
+        var promptParser = new PromptParser($('#import-field').val())
+        promptParser.parse()
+        if (promptParser.getElements().length == 0) throw 'Unknown error while parsing prompt'
+
+        console.log(promptParser.getElements())
+
+        promptTranslator = new PromptTranslator(promptParser.getElements())
+        promptTranslator.translate()
+
+        var objects = promptTranslator.getObjects()
+        $('#inputarea').text('')
+          objects.forEach((object) => {
+          object.append('<i class="fas fa-minus-circle"></i>')
+          $('#inputarea').append(object)
+          addEventListener(object)
+        })
+        updateOutput()
+      } catch (e) {
+        success = false
+        //@TODO: Make more beautiful alert
+        alert(e)
+      }
+    } else {
+      success = false
+    }
+  } else {
+    $('#import-field').val('')
+  }
+
+  if (success) $('#import-field').toggleClass('visible')
+})
+
+/**
+ * Parses the prompt into an array which can be interpreted easier.
+ * 
+ * Variables and methods, beginning with an underscore MUST NOT accessed from outside!
+ * 
+ * _MAX_ITERATIONS is a constant that defines, how often the parse() method can call itself,
+ * to prevent the browser to freeze because of too much recursion.
+ * If the string prompt is longer than this value, it can't be parsed.
+ */
+class PromptParser {
+
+  //@TODO: maybe higher?
+  _MAX_ITERATIONS = 2000
```

Comment:
Where did you come up with that number? Have you tested it?

Reply:
its a number, the browser can handle. _my_ browser. i'll change it to string.length. this should work in every case